Makefile

```diff
@@ -1,53 +1,78 @@
-.PHONY: clean test dist test-dist
+PROJ := portable8439
 
-src = $(wildcard src/*.c) $(wildcard src/chacha-portable/*.c) $(wildcard src/poly1305-donna/*.c)
+STD ?= -std=c99
+CFLAGS ?= -O3
+CFLAGS += $(STD) -Wpedantic -Wall -Wextra -Isrc -Isrc/chacha-portable -Isrc/poly1305-donna -fstack-protector
+LDFLAGS :=
+VERSION ?= dev-version
+PREFIX ?= /usr/local
 
-CFLAGS?=-O3
-CFLAGS+=-std=c99 -pedantic -Wall -Wextra -Isrc -Isrc/chacha-portable -Isrc/poly1305-donna -fstack-protector
-LDFLAGS = 
+SRCDIR := src
+BLDDIR := dist
+OBJDIR := $(BLDDIR)/obj
+TSTDIR := $(BLDDIR)/test
 
-VERSION?=dev-version
+SOURCES := $(shell find $(SRCDIR) -type f -iname '*.c')
+OBJECTS := $(patsubst $(SRCDIR)%, $(OBJDIR)%, $(patsubst %.c, %.o, $(SOURCES)))
```

Comment:
This setup will compile every c file separately and later linked together.

Compiling it in a single phase allows the c compiler to make faster code, so that should still be a goal here as well. 

Next to this, the algamized version only exports the final API, none of the internals, that is an important feature I would like to keep.

Reply:
This is really easy to fix. I've just had the libraries instead build from the “algamized” (is this similar to “amalgamated”? I've not actually been able to find other uses of this word elsewhere) source files.

Reply:
ah sorry, indeed, I meant amalgamated.